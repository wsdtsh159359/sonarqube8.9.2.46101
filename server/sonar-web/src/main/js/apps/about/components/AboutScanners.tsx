/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { Link } from 'react-router';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { getBaseUrl } from 'sonar-ui-common/helpers/urls';

const scanners = [
  {
    key: 'sonarqube',
    link: '/documentation/analysis/scan/sonarscanner/'
  },
  {
    key: 'msbuild',
    link: '/documentation/analysis/scan/sonarscanner-for-msbuild/'
  },
  {
    key: 'maven',
    link: '/documentation/analysis/scan/sonarscanner-for-maven/'
  },
  {
    key: 'gradle',
    link: '/documentation/analysis/scan/sonarscanner-for-gradle/'
  },
  {
    key: 'jenkins',
    link: '/documentation/analysis/scan/sonarscanner-for-jenkins/'
  },
  {
    key: 'ant',
    link: '/documentation/analysis/scan/sonarscanner-for-ant/'
  }
];

export default function AboutScanners() {
  return (
    <div className="boxed-group">
      <h2>{translate('about_page.scanners')}</h2>
      <div className="boxed-group-inner">
        <p className="about-page-text">{translate('about_page.scanners.text')}</p>
        <div className="about-page-analyzers">
          {scanners.map(scanner => (
            <Link
              className="about-page-analyzer-box"
              key={scanner.key}
              to={scanner.link}
              target="_blank">
              <img
                alt={translate('about_page.scanners', scanner.key)}
                height={60}
                src={`${getBaseUrl()}/images/scanner-logos/${scanner.key}.svg`}
              />
            </Link>
          ))}
        </div>
      </div>
    </div>
  );
}
