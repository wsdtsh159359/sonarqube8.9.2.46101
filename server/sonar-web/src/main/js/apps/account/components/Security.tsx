/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { Helmet } from 'react-helmet-async';
import { connect } from 'react-redux';
import { translate } from 'sonar-ui-common/helpers/l10n';
import ResetPasswordForm from '../../../components/common/ResetPasswordForm';
import { getCurrentUser, Store } from '../../../store/rootReducer';
import Tokens from './Tokens';

export interface SecurityProps {
  user: T.LoggedInUser;
}

export function Security({ user }: SecurityProps) {
  return (
    <div className="account-body account-container">
      <Helmet defer={false} title={translate('my_account.security')} />
      <Tokens login={user.login} />
      {user.local && (
        <section className="boxed-group">
          <h2 className="spacer-bottom">{translate('my_profile.password.title')}</h2>
          <ResetPasswordForm className="boxed-group-inner" user={user} />
        </section>
      )}
    </div>
  );
}

const mapStateToProps = (state: Store) => ({
  user: getCurrentUser(state) as T.LoggedInUser
});

export default connect(mapStateToProps)(Security);
