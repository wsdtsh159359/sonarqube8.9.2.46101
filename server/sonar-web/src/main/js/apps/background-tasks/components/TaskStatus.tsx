/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import PendingIcon from 'sonar-ui-common/components/icons/PendingIcon';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { TaskStatuses } from '../../../types/tasks';

interface Props {
  status: string;
}

export default function TaskStatus({ status }: Props) {
  let inner;

  switch (status) {
    case TaskStatuses.Pending:
      inner = <PendingIcon />;
      break;
    case TaskStatuses.InProgress:
      inner = <i className="spinner" />;
      break;
    case TaskStatuses.Success:
      inner = (
        <span className="badge badge-success">{translate('background_task.status.SUCCESS')}</span>
      );
      break;
    case TaskStatuses.Failed:
      inner = (
        <span className="badge badge-error">{translate('background_task.status.FAILED')}</span>
      );
      break;
    case TaskStatuses.Canceled:
      inner = <span className="badge">{translate('background_task.status.CANCELED')}</span>;
      break;
    default:
      inner = '';
  }

  return <td className="thin spacer-right">{inner}</td>;
}
