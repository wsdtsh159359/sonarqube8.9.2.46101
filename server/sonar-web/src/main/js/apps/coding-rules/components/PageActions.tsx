/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import ReloadButton from 'sonar-ui-common/components/controls/ReloadButton';
import DeferredSpinner from 'sonar-ui-common/components/ui/DeferredSpinner';
import { translate } from 'sonar-ui-common/helpers/l10n';
import PageCounter from '../../../components/common/PageCounter';
import PageShortcutsTooltip from '../../../components/ui/PageShortcutsTooltip';

export interface PageActionsProps {
  loading: boolean;
  onReload: () => void;
  paging?: T.Paging;
  selectedIndex?: number;
}

export default function PageActions(props: PageActionsProps) {
  return (
    <div className="display-flex-center">
      <PageShortcutsTooltip
        className="big-spacer-right"
        leftAndRightLabel={translate('issues.to_navigate')}
        upAndDownLabel={translate('coding_rules.to_select_rules')}
      />

      <DeferredSpinner loading={props.loading}>
        <ReloadButton onClick={props.onReload} />
      </DeferredSpinner>

      {props.paging && (
        <PageCounter
          className="spacer-left"
          current={props.selectedIndex}
          label={translate('coding_rules._rules')}
          total={props.paging.total}
        />
      )}
    </div>
  );
}
