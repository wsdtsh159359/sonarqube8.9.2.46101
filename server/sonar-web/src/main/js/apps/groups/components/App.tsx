/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { Helmet } from 'react-helmet-async';
import ListFooter from 'sonar-ui-common/components/controls/ListFooter';
import SearchBox from 'sonar-ui-common/components/controls/SearchBox';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { omitNil } from 'sonar-ui-common/helpers/request';
import { createGroup, deleteGroup, searchUsersGroups, updateGroup } from '../../../api/user_groups';
import Suggestions from '../../../app/components/embed-docs-modal/Suggestions';
import DeleteForm from './DeleteForm';
import Form from './Form';
import Header from './Header';
import List from './List';

interface State {
  groups?: T.Group[];
  editedGroup?: T.Group;
  groupToBeDeleted?: T.Group;
  loading: boolean;
  paging?: T.Paging;
  query: string;
}

export default class App extends React.PureComponent<{}, State> {
  mounted = false;
  state: State = { loading: true, query: '' };

  componentDidMount() {
    this.mounted = true;
    this.fetchGroups();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  makeFetchGroupsRequest = (data?: { p?: number; q?: string }) => {
    this.setState({ loading: true });
    return searchUsersGroups({
      q: this.state.query,
      ...data
    });
  };

  stopLoading = () => {
    if (this.mounted) {
      this.setState({ loading: false });
    }
  };

  fetchGroups = async (data?: { p?: number; q?: string }) => {
    try {
      const { groups, paging } = await this.makeFetchGroupsRequest(data);
      if (this.mounted) {
        this.setState({ groups, loading: false, paging });
      }
    } catch {
      this.stopLoading();
    }
  };

  fetchMoreGroups = async () => {
    const { paging: currentPaging } = this.state;
    if (currentPaging && currentPaging.total > currentPaging.pageIndex * currentPaging.pageSize) {
      try {
        const { groups, paging } = await this.makeFetchGroupsRequest({
          p: currentPaging.pageIndex + 1
        });
        if (this.mounted) {
          this.setState(({ groups: existingGroups = [] }) => ({
            groups: [...existingGroups, ...groups],
            loading: false,
            paging
          }));
        }
      } catch {
        this.stopLoading();
      }
    }
  };

  search = (query: string) => {
    this.fetchGroups({ q: query });
    this.setState({ query });
  };

  refresh = async () => {
    const { paging, query } = this.state;

    await this.fetchGroups({ q: query });

    // reload all pages in order
    if (paging && paging.pageIndex > 1) {
      for (let p = 1; p < paging.pageIndex; p++) {
        // eslint-disable-next-line no-await-in-loop
        await this.fetchMoreGroups(); // This is a intentional promise chain
      }
    }
  };

  closeDeleteForm = () => {
    this.setState({ groupToBeDeleted: undefined });
  };

  closeEditForm = () => {
    this.setState({ editedGroup: undefined });
  };

  openDeleteForm = (group: T.Group) => {
    this.setState({ groupToBeDeleted: group });
  };

  openEditForm = (group: T.Group) => {
    this.setState({ editedGroup: group });
  };

  handleCreate = async (data: { description: string; name: string }) => {
    await createGroup({ ...data });

    await this.refresh();
  };

  handleDelete = async () => {
    const { groupToBeDeleted } = this.state;

    if (!groupToBeDeleted) {
      return;
    }

    await deleteGroup({ name: groupToBeDeleted.name });

    await this.refresh();

    if (this.mounted) {
      this.setState({ groupToBeDeleted: undefined });
    }
  };

  handleEdit = async ({ name, description }: { name?: string; description: string }) => {
    const { editedGroup } = this.state;

    if (!editedGroup) {
      return;
    }

    const data = {
      description,
      id: editedGroup.id,
      // pass `name` only if it has changed, otherwise the WS fails
      ...omitNil({ name: name !== editedGroup.name ? name : undefined })
    };

    await updateGroup(data);

    if (this.mounted) {
      this.setState(({ groups = [] }: State) => ({
        editedGroup: undefined,
        groups: groups.map(group =>
          group.name === editedGroup.name ? { ...group, ...data } : group
        )
      }));
    }
  };

  render() {
    const { editedGroup, groupToBeDeleted, groups, loading, paging, query } = this.state;

    const showAnyone = 'anyone'.includes(query.toLowerCase());

    return (
      <>
        <Suggestions suggestions="user_groups" />
        <Helmet defer={false} title={translate('user_groups.page')} />
        <div className="page page-limited" id="groups-page">
          <Header onCreate={this.handleCreate} />

          <SearchBox
            className="big-spacer-bottom"
            id="groups-search"
            minLength={2}
            onChange={this.search}
            placeholder={translate('search.search_by_name')}
            value={query}
          />

          {groups !== undefined && (
            <List
              groups={groups}
              onDelete={this.openDeleteForm}
              onEdit={this.openEditForm}
              onEditMembers={this.refresh}
              showAnyone={showAnyone}
            />
          )}

          {groups !== undefined && paging !== undefined && (
            <div id="groups-list-footer">
              <ListFooter
                count={showAnyone ? groups.length + 1 : groups.length}
                loading={loading}
                loadMore={this.fetchMoreGroups}
                ready={!loading}
                total={showAnyone ? paging.total + 1 : paging.total}
              />
            </div>
          )}

          {groupToBeDeleted && (
            <DeleteForm
              group={groupToBeDeleted}
              onClose={this.closeDeleteForm}
              onSubmit={this.handleDelete}
            />
          )}

          {editedGroup && (
            <Form
              confirmButtonText={translate('update_verb')}
              group={editedGroup}
              header={translate('groups.update_group')}
              onClose={this.closeEditForm}
              onSubmit={this.handleEdit}
            />
          )}
        </div>
      </>
    );
  }
}
