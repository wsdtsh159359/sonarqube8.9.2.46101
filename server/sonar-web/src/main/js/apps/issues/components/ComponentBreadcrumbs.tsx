/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import QualifierIcon from 'sonar-ui-common/components/icons/QualifierIcon';
import { translateWithParameters } from 'sonar-ui-common/helpers/l10n';
import { collapsePath, limitComponentName } from 'sonar-ui-common/helpers/path';
import { getSelectedLocation } from '../utils';

interface Props {
  component?: T.Component;
  issue: T.Issue;
  selectedFlowIndex?: number;
  selectedLocationIndex?: number;
}

export default function ComponentBreadcrumbs({
  component,
  issue,
  selectedFlowIndex,
  selectedLocationIndex
}: Props) {
  const displayProject = !component || !['TRK', 'BRC', 'DIR'].includes(component.qualifier);
  const displaySubProject = !component || !['BRC', 'DIR'].includes(component.qualifier);

  const selectedLocation = getSelectedLocation(issue, selectedFlowIndex, selectedLocationIndex);
  const componentName = selectedLocation ? selectedLocation.componentName : issue.componentLongName;

  return (
    <div
      aria-label={translateWithParameters(
        'issues.on_file_x',
        `${displayProject ? issue.projectName + ', ' : ''}${componentName}`
      )}
      className="component-name text-ellipsis">
      <QualifierIcon className="spacer-right" qualifier={issue.componentQualifier} />

      {displayProject && (
        <span title={issue.projectName}>
          {limitComponentName(issue.projectName)}
          <span className="slash-separator" />
        </span>
      )}

      {displaySubProject && issue.subProject !== undefined && issue.subProjectName !== undefined && (
        <span title={issue.subProjectName}>
          {limitComponentName(issue.subProjectName)}
          <span className="slash-separator" />
        </span>
      )}

      <span title={componentName}>{collapsePath(componentName || '')}</span>
    </div>
  );
}
