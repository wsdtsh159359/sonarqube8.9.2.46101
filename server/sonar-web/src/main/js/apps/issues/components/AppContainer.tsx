/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { connect } from 'react-redux';
import { lazyLoadComponent } from 'sonar-ui-common/components/lazyLoadComponent';
import { searchIssues } from '../../../api/issues';
import { withRouter } from '../../../components/hoc/withRouter';
import { parseIssueFromResponse } from '../../../helpers/issues';
import { fetchBranchStatus } from '../../../store/rootActions';
import { getCurrentUser, Store } from '../../../store/rootReducer';
import { FetchIssuesPromise } from '../../../types/issues';

const IssuesAppContainer = lazyLoadComponent(() => import('./App'), 'IssuesAppContainer');

const mapStateToProps = (state: Store) => ({
  currentUser: getCurrentUser(state),
  fetchIssues
});

const fetchIssues = (query: T.RawQuery) => {
  return searchIssues({
    ...query,
    additionalFields: '_all',
    timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone
  }).then(response => {
    const parsedIssues = response.issues.map(issue =>
      parseIssueFromResponse(issue, response.components, response.users, response.rules)
    );
    return { ...response, issues: parsedIssues } as FetchIssuesPromise;
  });
};

const mapDispatchToProps = { fetchBranchStatus };

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(IssuesAppContainer));
