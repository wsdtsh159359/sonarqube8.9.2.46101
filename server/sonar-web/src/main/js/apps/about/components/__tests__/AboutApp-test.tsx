/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { addWhitePageClass, removeWhitePageClass } from 'sonar-ui-common/helpers/pages';
import { waitAndUpdate } from 'sonar-ui-common/helpers/testUtils';
import { searchProjects } from '../../../../api/components';
import { getFacet } from '../../../../api/issues';
import { mockCurrentUser, mockLocation } from '../../../../helpers/testMocks';
import { AboutApp } from '../AboutApp';
import EntryIssueTypes from '../EntryIssueTypes';

jest.mock('sonar-ui-common/helpers/pages', () => ({
  addWhitePageClass: jest.fn(),
  removeWhitePageClass: jest.fn()
}));

jest.mock('../../../../api/components', () => ({
  searchProjects: jest.fn().mockResolvedValue({ paging: { total: 5 } })
}));

jest.mock('../../../../api/issues', () => ({
  getFacet: jest.fn().mockResolvedValue({
    facet: [
      { count: 5, val: 'CODE_SMELL' },
      { count: 10, val: 'BUG' },
      { count: 0, val: 'VULNERABILITY' },
      { count: 5, val: 'SECURITY_HOTSPOT' }
    ]
  })
}));

jest.mock('../../../../app/components/GlobalContainer', () => ({
  default: class GlobalContainer extends React.Component {
    static displayName = 'GlobalContainer';
    render() {
      return this.props.children;
    }
  }
}));

it('should render correctly', async () => {
  const wrapper = shallowRender();
  await waitAndUpdate(wrapper);

  expect(wrapper).toMatchSnapshot();
  expect(addWhitePageClass).toBeCalled();

  wrapper.unmount();
  expect(removeWhitePageClass).toBeCalled();

  expect(
    shallowRender({
      indexationContext: {
        status: { isCompleted: false, percentCompleted: 10, hasFailures: false }
      }
    })
  ).toMatchSnapshot('when indexation not complete');
});

it('should load issues, projects, and custom text upon shallowing', () => {
  const fetchAboutPageSettings = jest.fn();
  shallowRender({ fetchAboutPageSettings });
  expect(fetchAboutPageSettings).toBeCalled();
  expect(searchProjects).toBeCalled();
  expect(getFacet).toBeCalled();
});

it('should not display issues if the WS return an http error', async () => {
  (getFacet as jest.Mock).mockRejectedValueOnce(undefined);

  const wrapper = shallowRender();

  await waitAndUpdate(wrapper);

  expect(wrapper.find(EntryIssueTypes).exists()).toBe(false);
});

function shallowRender(props: Partial<AboutApp['props']> = {}) {
  return shallow(
    <AboutApp
      currentUser={mockCurrentUser()}
      customText="Lorem ipsum"
      fetchAboutPageSettings={jest.fn().mockResolvedValue('')}
      location={mockLocation()}
      indexationContext={{
        status: { isCompleted: true, percentCompleted: 100, hasFailures: false }
      }}
      {...props}
    />
  );
}
