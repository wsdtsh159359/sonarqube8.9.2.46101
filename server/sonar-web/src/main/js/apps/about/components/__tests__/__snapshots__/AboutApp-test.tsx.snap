// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="page page-limited about-page"
  id="about-page"
>
  <A11ySkipTarget
    anchor="about_main"
  />
  <div
    className="about-page-entry"
  >
    <div
      className="about-page-intro"
    >
      <h1
        className="big-spacer-bottom"
      >
        layout.sonar.slogan
      </h1>
      <Link
        className="button button-active big-spacer-right"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/sessions/new"
      >
        layout.login
      </Link>
      <Link
        className="button"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/documentation"
      >
        about_page.read_documentation
      </Link>
    </div>
    <div
      className="about-page-instance"
    >
      <AboutProjects
        count={5}
        loading={false}
      />
      <EntryIssueTypes
        bugs={10}
        codeSmells={5}
        loading={false}
        vulnerabilities={0}
      />
    </div>
  </div>
  <div
    className="about-page-section"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Lorem ipsum",
      }
    }
  />
  <AboutLanguages />
  <AboutQualityModel />
  <div
    className="flex-columns"
  >
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutCleanCode />
    </div>
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutLeakPeriod />
    </div>
  </div>
  <div
    className="flex-columns"
  >
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutQualityGates />
    </div>
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutStandards />
    </div>
  </div>
  <AboutScanners />
</div>
`;

exports[`should render correctly: when indexation not complete 1`] = `
<div
  className="page page-limited about-page"
  id="about-page"
>
  <A11ySkipTarget
    anchor="about_main"
  />
  <div
    className="about-page-entry"
  >
    <div
      className="about-page-intro"
    >
      <h1
        className="big-spacer-bottom"
      >
        layout.sonar.slogan
      </h1>
      <Link
        className="button button-active big-spacer-right"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/sessions/new"
      >
        layout.login
      </Link>
      <Link
        className="button"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/documentation"
      >
        about_page.read_documentation
      </Link>
    </div>
    <div
      className="about-page-instance"
    >
      <AboutProjects
        count={0}
        loading={true}
      />
    </div>
  </div>
  <div
    className="about-page-section"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Lorem ipsum",
      }
    }
  />
  <AboutLanguages />
  <AboutQualityModel />
  <div
    className="flex-columns"
  >
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutCleanCode />
    </div>
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutLeakPeriod />
    </div>
  </div>
  <div
    className="flex-columns"
  >
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutQualityGates />
    </div>
    <div
      className="flex-column flex-column-half about-page-group-boxes"
    >
      <AboutStandards />
    </div>
  </div>
  <AboutScanners />
</div>
`;
